pkg/services/ngalert/store/alert_rule.go

```diff
@@ -156,7 +158,7 @@ func (st DBstore) UpsertAlertRules(rules []UpsertRule) error {
 				existingAlertRule, err := getAlertRuleByUID(sess, r.New.UID, r.New.OrgID)
 				if err != nil {
 					if errors.Is(err, ngmodels.ErrAlertRuleNotFound) {
-						return nil
+						return fmt.Errorf("err: %w, UID: %s", err, r.New.UID)
```

Comment:
Is this correct? If we rewrite this error, this will no longer match in an `errors.Is(err, ngmodels.ErrAlertRuleNotFound)`?

Reply:
I think it is since I use %w

Reply:
would it be more consistent to use errors.Wrap and then?

Reply:
Do you suggest using [github.com/pkg/errors](https://pkg.go.dev/github.com/pkg/errors) instead of [build-in errors](https://golang.org/pkg/errors/)?
I'm fine with that change but given that so far it's mostly used only inside the `ngalert` module I wonder if it's a deliberate preference in grafana over the built-in one.